core/pv/src/main/java/org/phoebus/pv/pva/PVAStructureHelper.java

```diff
@@ -203,6 +215,45 @@ else if (column instanceof PVAStringArray)
                 types.add(String.class);
                 values.add(Arrays.asList(typed.get()));
             }
+            else if (column instanceof PVAShortArray)
+            {
+                final PVAShortArray typed = (PVAShortArray)column;
+                types.add(Short.TYPE);
+                if (typed.isUnsigned())
+                    values.add(ArrayUShort.of(typed.get()));
+                else
+                    values.add(ArrayShort.of(typed.get()));
+            }
+            else if (column instanceof PVALongArray)
+            {
+                final PVALongArray typed = (PVALongArray)column;
+                types.add(Long.TYPE);
+                if (typed.isUnsigned())
+                    values.add(ArrayULong.of(typed.get()));
+                else
+                    values.add(ArrayLong.of(typed.get()));
+            }
+            else if (column instanceof PVAByteArray)
+            {
+                final PVAByteArray typed = (PVAByteArray)column;
+                types.add(Byte.TYPE);
+                if (typed.isUnsigned())
+                    values.add(ArrayUByte.of(typed.get()));
+                else
+                    values.add(ArrayByte.of(typed.get()));
+            }
+            else if (column instanceof PVABoolArray)
+            {
+                final PVABoolArray typed = (PVABoolArray)column;
+                types.add(Boolean.TYPE);
+                boolean[] data = typed.get();
+                // Convert to boxed Integer to add to List
+                values.add(range(0, data.length).mapToObj(i -> data[i]).collect(toList()));
+            }
+            else 
+            {
+            	throw new RuntimeException("Could not decode table column of type: " + column.getClass());
```

Comment:
Yes `RuntimeException` is probably not the best, perhaps `IllegalArgumentException` is more appropriate? Although that is still a `RuntimeException`. Or are is there a suitable domain specific exception that already exists? 

When you say the "helper" what does that refer to? This method looks unchanged since it was first committed.
Something similar to what `decodeNTArrayField` and `decodeNTArray` do?

Reply:
I've had a look at this but I'm not sure how to convert it suitably. The whole column (`PVData`) could be converted into a `VString` but this doesn't really help as the structure added to the `VTable` needs to be a list-like object with an entry for each row.

Reply:
> I've had a look at this but I'm not sure how to convert it suitably.

OK, then throwing an exception is likely the best way to handle this.